git/remote.py

```diff
@@ -570,21 +570,36 @@ def _get_fetch_info_from_stderr(self, proc, progress):
 
         progress_handler = progress.new_message_handler()
 
+        error_message = None
+        stderr_text = None
+
         for line in proc.stderr:
             line = force_text(line)
             for pline in progress_handler(line):
                 if line.startswith('fatal:') or line.startswith('error:'):
-                    raise GitCommandError(("Error when fetching: %s" % line,), 2)
+                    error_message = "Error when fetching: %s" % (line,)
+                    break
```

Comment:
Are you sure it' s enough to break after the first line indicating a problem ? My intuition is that the lines that follow on stderr contain context, that you probably want to catch as well.